### ISSUE TITLE
Adding references does not seem to work.

### ISSUE THREAD

**Issue by michielvandijk** [rnuske/komaletter#15]
I wanted to add a few references in a submission letter for a paper but I am getting an error.

I am adding the standard information for making references:

In the yaml:
bibliography: c:/data/references.bib

In the text:
[@vandijk2021]

but get the following error:

! LaTeX Error: Environment CSLReferences undefined.

I noticed that some other people had similar problems in other cases and the problem seems to be related with Pandoc, for example see: https://github.com/crsh/prereg/issues/16. 

I hope you can have a look.

Session info:
R version 4.0.5 (2021-03-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] komaletter_0.4.2 compiler_4.0.5   fastmap_1.1.0    htmltools_0.5.2  tools_4.0.5      yaml_2.2.1       rmarkdown_2.11   knitr_1.36      
 [9] xfun_0.28        digest_0.6.28    rlang_0.4.12     evaluate_0.14

**Comment by michielvandijk** [comment]
By the way, I am using pandoc version: '2.11.4' if that would be useful. In combination with MikTex.

**Comment by rnuske** [comment]
Thanks for bringing this issue to my attention. I consider this a mixture of a bug report and a feature wish. As far as I know, nobody tried yet to produce an automated reference list within a komaletter. I can see that it could be useful for example in communicating a revision of a manuscript.

I first checked that the whished feature works in the underlying `rmarkdown::pdf_document()` which `komaletter` is build upon. And yes it works. Save the following in a file and compile it with `rmarkdown::render('filename')` or via the RStudio knit button.

```
---
title: "Test References"
bibliography: "`r knitr::write_bib('komaletter', 'refs.bib'); paste0('refs.bib')`" 
output: pdf_document
---

The komaletter package [@R-komaletter] is a cornucopia. It can already do a lot, but not yet everything. Reference lists have not been requested, so far. Let's see how we can implement this.

End of regular text.
```

So it should be possible to implement the 'reference list'-feature in `komaletter`. 

<br>

Now to the bug part: The handling of citations changed in pandoc 2.11. Pandoc now needs the "Environment CSLReferences" which has to be defined in the template. `rmarkdown::pdf_document()` uses pandoc's default latex template that contains the  CSLReferences Environment. `komaletter` uses its own lovingly handcrafted latex template. So the CSLReferences Environment must be added. I think something along the lines of https://github.com/rstudio/tufte/pull/89 would be sensible.

<br>

Leading to an open  **question** @michielvandijk:

The "Test References" example above shows that the refences are just added as a new paragraph to the end of the document when using `pdf_document()`. No headline, no extra vertical space, no nothing. So, where should we put the list of references in a letter and how to create a headline (the employed latex class `scrlttr2` does not support `\section, \subsection, ...` cf. #13) .

**Comment by rnuske** [comment]
Bug part fixed. Questions still open.

Install branch `refs` of `komaletter` (`remotes::install_github("rnuske/komaletter", ref="refs")`) and try to compile e.g. the following letter

```
---
author: Max Mustermann
return-address: 
  - Musterstr. 12
  - 34567 Musterstadt
address:
  - Robert Nuske
  - Somestreet 101
  - 10024 Somewhere
  
subject: Some nonsensical Subject
opening: Dear Robert,
closing: Best regards,
signature: Max Mustermann

bibliography: "`r knitr::write_bib('komaletter', 'refs.bib'); paste0('refs.bib')`" 

output: komaletter::komaletter
---

The komaletter package [@R-komaletter] is a cornucopia. It can already do a lot, but not yet everything. Reference lists have not been requested, so far. Let's see how we can implement this.

**References**
```

**Comment by michielvandijk** [comment]
Thanks for addressing this. I tested the example above and confirm it works. I do get a warning though:

Output created: test.pdf
Warning message:
In utils::citation(..., lib.loc = lib.loc) :
  no date field in DESCRIPTION file of package 'komaletter'

However, when I use the standard reference to a .bib list (bibliography: reference_list.bib) there is no warning so that is fine.

Regarding the open questions. 
Where to put it? I would argue that in nearly all cases when referenced are added to a letter, they will be added all the way at the end, like an annex. I guess that if the list of references is very long, it might be nicer to put the closing before the list of references but I foresee that this might be tricky as the header **References** need to be added manually, which will make it overly complex in combination with adding an automatic closing as specified in the YAML. Hence, I think the present solution is good enough. 

How to style it? I think the current approach (no additional layout) is fine. As a test I added `\tiny` just after the reference header and this results in a small font reference list, which sometimes might be convenient. I suspect that other Latex code for styling such as an additional line break, double-spacing, etc will also work but I have not tested this. Regarding the headline, manually adding a **References** header has my preference as this gives most flexibility.
